src/libcore/marker.rs

```diff
@@ -606,7 +607,12 @@ unsafe impl<T: ?Sized> Freeze for &mut T {}
 /// See the [`pin module`] documentation for more information on pinning.
 ///
 /// Implementing this trait lifts the restrictions of pinning off a type,
-/// which then allows it to move out with functions such as [`replace`].
+/// which then allows it to move out with functions such as [`mem::replace`].
+///
+/// `Unpin` has no consequence at all for non-pinned data. In particular,
+/// [`mem::replace`] will happily move `!Unpin` data. However, you cannot use
```

Comment:
I think that would just make things take more space without helping. `mem::replace` doesn't even mention any trait bound, so it would be really weird if it had any interaction with `Unpin`. I will add an explanation of this.

Every interruption of the text, like such an example, makes the flow of the text harder to follow-- so it must be worth it in terms of extra information content. (That'd be different if we moved all the examples to a separate section, but I don't consider that useful either.)

Reply:
I expanded the existing example a little, that should also help with this I think.